File: app/src/main/java/com/unnamed/b/atv/sample/fragment/FolderStructureFragment.java
package com.unnamed.b.atv.sample.fragment;

import android.app.Fragment;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import com.unnamed.b.atv.model.TreeNode;
import com.unnamed.b.atv.sample.R;
import com.unnamed.b.atv.sample.holder.IconTreeItemHolder;
import com.unnamed.b.atv.view.AndroidTreeView;

/**
 * Created by Bogdan Melnychuk on 2/12/15.
 */
public class FolderStructureFragment extends Fragment {
    private TextView statusBar;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_default, null, false);
        ViewGroup containerView = (ViewGroup) rootView.findViewById(R.id.container);

        statusBar = (TextView) rootView.findViewById(R.id.status_bar);

        TreeNode root = TreeNode.root();
        TreeNode computerRoot = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_laptop, "My Computer"));

        TreeNode myDocuments = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_folder, "My Documents"));
        TreeNode downloads = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_folder, "Downloads"));
        TreeNode file1 = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_drive_file, "File 1"));
        TreeNode file2 = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_drive_file, "File 2"));
        TreeNode file3 = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_drive_file, "File 3"));
        TreeNode file4 = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_drive_file, "File 4"));
        fillDownloadsFolder(downloads);
        downloads.addChildren(file1, file2, file3, file4);

        TreeNode myMedia = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_photo_library, "Photos"));
        TreeNode photo1 = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_photo, "Photo 1"));
        TreeNode photo2 = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_photo, "Photo 2"));
        TreeNode photo3 = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_photo, "Photo 3"));
        myMedia.addChildren(photo1, photo2, photo3);

        myDocuments.addChild(downloads);
        computerRoot.addChildren(myDocuments, myMedia);

        root.addChildren(computerRoot);

        AndroidTreeView tView = new AndroidTreeView(getActivity(), root);
        tView.setDefaultContainerStyle(R.style.TreeNodeStyleCustom);
        tView.setDefaultViewHolder(IconTreeItemHolder.class);
        tView.setDefaultNodeClickListener(nodeClickListener);

        containerView.addView(tView.getView());
        tView.expandNode(computerRoot);

        return rootView;
    }

    private static int counter = 0;

    private void fillDownloadsFolder(TreeNode node) {
        TreeNode downloads = new TreeNode(new IconTreeItemHolder.IconTreeItem(R.string.ic_folder, "Downloads" + (counter++)));
        node.addChild(downloads);
        if (counter < 5) {
            fillDownloadsFolder(downloads);
        }
    }

    private TreeNode.TreeNodeClickListener nodeClickListener = new TreeNode.TreeNodeClickListener() {
        @Override
        public void onClick(TreeNode node, Object value) {
            IconTreeItemHolder.IconTreeItem item = (IconTreeItemHolder.IconTreeItem) value;
            statusBar.setText("Last clicked: " + item.text);
            if (node.isLeaf()) {
                Toast.makeText(getActivity(), item.text, Toast.LENGTH_SHORT).show();
            }
        }
    };
}

File: app/src/main/java/com/unnamed/b/atv/sample/fragment/FolderStructureFragment.kt
package com.unnamed.b.atv.sample.fragment

import android.os.Bundle
import android.support.v4.app.Fragment
import android.text.TextUtils
import android.view.*
import android.widget.TextView
import android.widget.Toast
import com.unnamed.b.atv.model.TreeNode
import com.unnamed.b.atv.sample.R
import com.unnamed.b.atv.sample.holder.IconTreeItemHolder
import com.unnamed.b.atv.view.AndroidTreeView


/**
 * Created by Bogdan Melnychuk on 2/12/15.
 * Converted to Kolin by Kumar Shivang on 16/07/17
 */
class FolderStructureFragment : Fragment() {
    private var statusBar: TextView? = null
    private var tView: AndroidTreeView? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        val rootView = inflater.inflate(R.layout.fragment_default, container, false)
        val containerView = rootView.findViewById(R.id.container) as ViewGroup

        statusBar = rootView.findViewById(R.id.status_bar) as TextView

        val root = TreeNode.root()
        val computerRoot = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_laptop, "My Computer"))

        val myDocuments = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_folder, "My Documents"))
        val downloads = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_folder, "Downloads"))
        val file1 = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_drive_file, "Folder 1"))
        val file2 = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_drive_file, "Folder 2"))
        val file3 = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_drive_file, "Folder 3"))
        val file4 = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_drive_file, "Folder 4"))
        fillDownloadsFolder(downloads)
        downloads.addChildren(file1, file2, file3, file4)

        val myMedia = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_photo_library, "Photos"))
        val photo1 = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_photo, "Folder 1"))
        val photo2 = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_photo, "Folder 2"))
        val photo3 = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_photo, "Folder 3"))
        myMedia.addChildren(photo1, photo2, photo3)

        myDocuments.addChild(downloads)
        computerRoot.addChildren(myDocuments, myMedia)

        root.addChildren(computerRoot)

        tView = AndroidTreeView(activity, root)
        tView!!.setDefaultAnimation(true)
        tView!!.setDefaultContainerStyle(R.style.TreeNodeStyleCustom)
        tView!!.setDefaultViewHolder(IconTreeItemHolder::class.java)
        tView!!.setDefaultNodeClickListener(nodeClickListener)
        tView!!.setDefaultNodeLongClickListener(nodeLongClickListener)

        containerView.addView(tView!!.view)

        if (savedInstanceState != null) {
            val state = savedInstanceState.getString("tState")
            if (!TextUtils.isEmpty(state)) {
                tView!!.restoreState(state!!)
            }
        }

        return rootView
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        inflater.inflate(R.menu.menu, menu)
        super.onCreateOptionsMenu(menu, inflater)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.expandAll -> tView!!.expandAll()

            R.id.collapseAll -> tView!!.collapseAll()
        }
        return true
    }

    private var counter = 0

    private fun fillDownloadsFolder(node: TreeNode) {
        val downloads = TreeNode(IconTreeItemHolder.IconTreeItem(R.string.ic_folder, "Downloads" + counter++))
        node.addChild(downloads)
        if (counter < 5) {
            fillDownloadsFolder(downloads)
        }
    }

    private val nodeClickListener = object : TreeNode.TreeNodeClickListener {
        override fun onClick(node: TreeNode, value: Any) {
            val item = value as IconTreeItemHolder.IconTreeItem
            statusBar!!.text = "Last clicked: " + item.text
        }
    }

    private val nodeLongClickListener = object : TreeNode.TreeNodeLongClickListener {
        override fun onLongClick(node: TreeNode, value: Any): Boolean {
            val item = value as IconTreeItemHolder.IconTreeItem
            Toast.makeText(activity, "Long click: " + item.text, Toast.LENGTH_SHORT).show()
            return true
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putString("tState", tView!!.saveState)
    }
}
